account-kit/smart-contracts/src/ma-v2/account/nativeSMASigner.ts

```diff
@@ -0,0 +1,123 @@
+import type { SmartAccountSigner } from "@aa-sdk/core";
+import {
+  hashMessage,
+  hashTypedData,
+  type Hex,
+  type SignableMessage,
+  type TypedData,
+  type TypedDataDefinition,
+  type Chain,
+  type Address,
+} from "viem";
+
+import { packUOSignature, pack1271Signature } from "../utils.js";
+/**
+ * Creates an object with methods for generating a dummy signature, signing user operation hashes, signing messages, and signing typed data.
+ *
+ * @example
+ * ```ts
+ * import { singleSignerMessageSigner } from "@account-kit/smart-contracts";
+ 
+ * import { LocalAccountSigner } from "@aa-sdk/core";
+ *
+ * const MNEMONIC = "...":
+ *
+ * const account = createSMAV2Account({ config });
+ *
+ * const signer = LocalAccountSigner.mnemonicToAccountSigner(MNEMONIC);
+ *
+ * const messageSigner = singleSignerMessageSigner(signer, chain);
+ * ```
+ *
+ * @param {TSigner} signer Signer to use for signing operations
+ * @param {Chain} chain Chain object for the signer
+ * @param {Address} accountAddress address of the smart account using this signer
+ * @param {number} entityId the entity id of the signing validation
+ * @returns {object} an object with methods for signing operations and managing signatures
+ */
+export const nativeSMASigner = <TSigner extends SmartAccountSigner>(
+  signer: TSigner,
+  chain: Chain,
+  accountAddress: Address,
+  entityId: number
+) => {
```

Comment:
As mentioned in another PR: you might consider explicitly defining the return type here, which I think is `SmartAccountSigner<TSigner>` with a few extra methods, assuming you add in the missing signer fields `signerName` and `inner` or something like that.

But if you don't explicitly state the return type, there's no need for the `TSigner` generic parameter. Can be more simply written as
```ts
export const nativeSMASigner = (
  signer: SmartAccountSigner,
  // ...
) => {
  // ...
```